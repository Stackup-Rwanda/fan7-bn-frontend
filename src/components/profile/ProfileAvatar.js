import React from 'react';
import PropTypes from 'prop-types';
import './styles/profileAvatar.scss';
import profileImg from '../../assets/icons/icons8-customer-48.png';

const ProfileAvatar = ({ src }) => (
  <img className="avatar" src={src} alt="Profile Avatar" />
);

ProfileAvatar.propTypes = {
  src: PropTypes.string,
};

ProfileAvatar.defaultProps = {
  src: profileImg,
};

export default ProfileAvatar;
