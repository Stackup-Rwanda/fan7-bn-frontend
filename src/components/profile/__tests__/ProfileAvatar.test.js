import React from 'react';
import { mount, shallow } from 'enzyme';
import ProfileAvatar from '../ProfileAvatar';

describe('ProfileAvatar Component Test', () => {
  it('should return true if component exists', () => {
    const wrapper = mount(<ProfileAvatar />);

    expect(wrapper.find('img').exists()).toBe(true);
  });
});

describe('<ProfileAvatar /> shallow rendering tests', () => {
  it('matches the snapshot', () => {
    const tree = shallow(<ProfileAvatar />);
    expect(tree).toMatchSnapshot();
  });
});
