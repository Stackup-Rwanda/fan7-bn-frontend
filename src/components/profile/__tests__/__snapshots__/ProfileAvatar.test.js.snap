// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ProfileAvatar /> shallow rendering tests matches the snapshot 1`] = `
<img
  alt="Profile Avatar"
  className="avatar"
  src={
    Object {
      "process": [Function],
    }
  }
/>
`;
